cdap-metrics-sidecar/src/test/java/io/cdap/cdap/jmx/metrics/JMXMetricsCollectorTest.java

```
@@ -0,0 +1,144 @@
+import javax.management.MBeanServer;
+import javax.management.remote.JMXConnectorServer;
+import javax.management.remote.JMXConnectorServerFactory;
+import javax.management.remote.JMXServiceURL;
+
+import static org.mockito.Matchers.anyLong;
+import static org.mockito.Matchers.eq;
+import static org.mockito.Mockito.atLeast;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class JMXMetricsCollectorTest {
+  private static final int SERVER_PORT = 11023;
+  private static final String SERVICE_NAME = "test-service";
+  private static JMXConnectorServer svr;
+
+  private static final Map<String, String> METRICS_CONTEXT = ImmutableMap.of(
+    Constants.Metrics.Tag.NAMESPACE, NamespaceId.SYSTEM.getNamespace(),
+    Constants.Metrics.Tag.COMPONENT, SERVICE_NAME);
+
+  @Mock
+  private MetricsCollectionService mockMetricsService;
+  @Mock
+  private MetricsContext mockContext;
+  @Mock
+  private Environment mockEnv;
+
+  @Before
+  public void beforeEach() {
+    MockitoAnnotations.initMocks(this);
+    when(mockMetricsService.getContext(METRICS_CONTEXT)).thenReturn(mockContext);
+  }
+
+  @BeforeClass
+  public static void setupClass() throws IOException {
+    svr = createJMXConnectorServer(SERVER_PORT);
+    svr.start();
+    Assert.assertEquals(svr.isActive(), true);
+  }
+
+  @AfterClass
+  public static void teardownClass() throws IOException {
+    svr.stop();
+  }
+
+  @Test(expected = IllegalArgumentException.class)
+  public void testInvalidPortInConfig() throws Exception {
+    when(mockEnv.getVariable(Constants.JMXMetricsCollector.COMPONENT_NAME_ENV_VAR)).thenReturn(SERVICE_NAME);
+    CConfiguration cConf = CConfiguration.create();
+    cConf.setInt(Constants.JMXMetricsCollector.SERVER_PORT, -1);
+    cConf.setInt(Constants.JMXMetricsCollector.POLL_INTERVAL, 100);
+    JMXMetricsCollector jmxMetrics = new JMXMetricsCollector(cConf, mockMetricsService, mockEnv);
+    jmxMetrics.start();
```

constructor itself should throw the exception here, so start() and stop() call can be removed.

---

You're right, removed.